doc/read-the-docs-site/howtos/exporting-a-blueprint.rst

```diff
@@ -0,0 +1,403 @@
+Haskell type from which its derived with a corresponding annotation:
+
+* ``SchemaTitle`` 
+* ``SchemaDescription``
+* ``SchemaComment``
+
+For example, to add a title and description to the ``MyParams`` type,
+we can use the ``SchemaTitle`` and ``SchemaDescription`` annotations:
+
+.. literalinclude:: Cip57Blueprint.hs
+    :start-after: BEGIN MyParams annotations
+    :end-before:  END MyParams annotations
+
+results in the following JSON schema definition:
+
+.. code-block:: json
+
+  {
+    "title": "Title for the MyParams definition",
+    "description": "Description for the MyParams definition",
+    "dataType": "constructor",
+    "fields": [
+      { "$ref": "#/definitions/Bool" },
+      { "$ref": "#/definitions/Integer" }
+    ],
+    "index": 0
+  }
+
+For sum-types its possible to annotate constructors:
+
+.. literalinclude:: Cip57Blueprint.hs
+    :start-after: BEGIN MyRedeemer annotations
+    :end-before:  END MyRedeemer annotations
+
+to produce the JSON schema definition:
+
+.. code-block:: json
+
+  {
+    "oneOf": [
+      {
+        "$comment": "Left redeemer",
+        "dataType": "constructor",
+        "fields": [],
+        "index": 0
+      },
+      {
+        "$comment": "Right redeemer",
+        "dataType": "constructor",
+        "fields": [],
+        "index": 1
+      }
+    ]
+  }
+
+It is also possible to annotate validator's parameter or argument **type** 
+(as opposed to annotating *constructors*):
+
+.. code-block:: haskell
+
+  {-# ANN type MyParams (SchemaTitle "Example parameter title") #-}
+  {-# ANN type MyRedeemer (SchemaTitle "Example redeemer title") #-}
+
+and then instead of providing them literally
+
+.. code-block:: haskell
+
+  myValidator =
+    MkValidatorBlueprint
+      { ... elided
+      , validatorParameters =
+          [ MkParameterBlueprint
+              { parameterTitle = Just "My Validator Parameters"
+              , parameterDescription = Just "Compile-time validator parameters"
+              , parameterPurpose = Set.singleton Spend
+              , parameterSchema = definitionRef @MyParams
+              }
+          ]
+      , validatorRedeemer =
+          MkArgumentBlueprint
+            { argumentTitle = Just "My Redeemer"
+            , argumentDescription = Just "A redeemer that does something awesome"
+            , argumentPurpose = Set.fromList [Spend, Mint]
+            , argumentSchema = definitionRef @MyRedeemer
+            }
+      , ... elided
+      }
+
+use TH to have a more concise version :
+
+.. code-block:: haskell
+
+  myValidator =
+    MkValidatorBlueprint
+      { ... elided
+      , validatorParameters =
+          [ $(deriveParameterBlueprint ''MyParams (Set.singleton Purpose.Spend)) ]
+      , validatorRedeemer =
+          $(deriveArgumentBlueprint ''MyRedeemer (Set.fromList [Purpose.Spend, Purpose.Mint]))
+      , ... elided
+      }
+
+
+Result
+------
+
+Here is the full `CIP-0057`_ blueprint produced by this "howto" example:
```

Comment:
you could check this in as a standalone file and have a test that checks that it's actually the blueprint you get from the haskell file.

Reply:
it looks like it's already checked in separately?

Reply:
Last time I tried to do this I hit this https://github.com/input-output-hk/iogx/pull/45/files

but now as it has been merged I'll try it again (after updating iogx)